### PR TITLE
catalog: Simplify GetServicesForServiceAccount()

### PR DESCRIPTION
**Pull request by draychev**
In this PR:
 - remove a level of indentation by using `continue` in the `constants.AzureProviderName` check
 - variable `service` conflicts with imported package - renamed to `svc` 
 - simplified slice allocation and returning of empty slice

No functional code changes in this PR.

Please mark with X for applicable areas.

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ x ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests / CI System      [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

### CODE REVIEW

File: pkg/catalog/service.go [hunk 1]
```diff
@@ -7,23 +7,27 @@ import (
...
-				services = append(services, service)
-			}
+		if provider.GetID() == constants.AzureProviderName {
+			continue
+		}
+
+		log.Trace().Msgf("[%s] Looking for Services for Name=%s", provider.GetID(), sa)
+		svc, err := provider.GetServiceForServiceAccount(sa)
+		if err != nil {
+			log.Warn().Msgf("Error getting services from provider %s: %s", provider.GetID(), err)
```
Comment (shashankram) [inline pkg/catalog/service.go:21]: Can we log the ServiceAccount in these 2 log messages?

Comment (draychev) [reply pkg/catalog/service.go]: Thanks for bringing this up - these log messages indeed need some work.

**Comment by draychev** [comment]
Merged from main, removed one log message (seems redundand), rewrote the log messages for clarity.
PTAL

File: pkg/catalog/service.go [hunk 2]
```diff
@@ -9,21 +11,27 @@ import (
...
 		// TODO (#88) : remove this provider check once we have figured out the service account story for azure vms
-		if provider.GetID() != constants.AzureProviderName {
-			log.Trace().Msgf("[%s] Looking for Services for Name=%s", provider.GetID(), sa)
-			providerServices, err := provider.GetServicesForServiceAccount(sa)
-			if err != nil {
-				log.Warn().Msgf("Error getting services from provider %s: %s", provider.GetID(), err)
-			} else {
-				log.Trace().Msgf("[%s] Found services %v for Name=%s", provider.GetID(), services, sa)
-				services = append(services, providerServices...)
+		if provider.GetID() == constants.AzureProviderName {
```
Comment (eduser25) [inline pkg/catalog/service.go:16]: Should we just remove AzureProvider from `endpointProviders` for the time being instead?

I think it'll look cleaner overall, and once we retake the VM endpoints effort we can always add it back

Comment (draychev) [reply pkg/catalog/service.go:16]: @eduser25 - we could inded remove the Azure Endpoint Provider stuff until we have Identity + VM representation in SMI.

**Pull request merged**